server/src/test/java/org/elasticsearch/index/mapper/BinaryRangeUtilTests.java

```diff
@@ -140,6 +146,61 @@ public void testEncode_Float() {
         }
     }
 
+    public void testDecodeLong() {
+        long[] cases = new long[] { Long.MIN_VALUE, -2049, -2048, -128, -3, -1, 0, 1, 3, 125, 2048, 2049, Long.MAX_VALUE};
+        for (long expected : cases) {
+            byte[] encoded = BinaryRangeUtil.encodeLong(expected);
+            int offset = 0;
+            int length = BinaryRangeUtil.LengthType.VARIABLE.readLength(encoded, offset);
+            assertEquals(expected, BinaryRangeUtil.decodeLong(encoded, offset,  length));
+        }
+    }
+
+    public void testDecodeLongRanges() throws IOException {
+        long start = randomLong();
+        long end = randomLongBetween(start + 1, Long.MAX_VALUE);
+        RangeFieldMapper.Range expected = new RangeFieldMapper.Range(RangeFieldMapper.RangeType.LONG, start, end, true, true);
+        List<RangeFieldMapper.Range> decoded = BinaryRangeUtil.decodeLongRanges(BinaryRangeUtil.encodeLongRanges(singleton(expected)));
+        assertEquals(1, decoded.size());
+        RangeFieldMapper.Range actual = decoded.get(0);
+        assertEquals(expected, actual);
+    }
+
+    public void testDecodeDoubleRanges() throws IOException {
+        double start = randomDouble();
+        double end = randomDoubleBetween(Math.nextUp(start), Double.MAX_VALUE, false);
+        RangeFieldMapper.Range expected = new RangeFieldMapper.Range(RangeFieldMapper.RangeType.DOUBLE, start, end, true, true);
+        List<RangeFieldMapper.Range> decoded = BinaryRangeUtil.decodeDoubleRanges(BinaryRangeUtil.encodeDoubleRanges(singleton(expected)));
+        assertEquals(1, decoded.size());
+        RangeFieldMapper.Range actual = decoded.get(0);
+        assertEquals(expected, actual);
+    }
+
+    public void testDecodeFloatRanges() throws IOException {
+        float start = randomFloat();
+        // for some reason, ESTestCase doesn't provide randomFloatBetween
+        float end = randomFloat();
+        if (start > end) {
+            float temp = start;
+            start = end;
+            end = temp;
+        }
+        RangeFieldMapper.Range expected = new RangeFieldMapper.Range(RangeFieldMapper.RangeType.FLOAT, start, end, true, true);
+        List<RangeFieldMapper.Range> decoded = BinaryRangeUtil.decodeFloatRanges(BinaryRangeUtil.encodeFloatRanges(singleton(expected)));
+        assertEquals(1, decoded.size());
+        RangeFieldMapper.Range actual = decoded.get(0);
+        assertEquals(expected, actual);
+    }
+
+    public void testDecodeIPRanges() throws IOException {
```

Comment:
should we test IPv6 addresses too ?

Reply:
I added one example IPv6 address.  I don't feel great about the lack of randomization here, but I feel like adding random generation of valid IP addresses is out of scope for this PR.

Reply:
agreed, thanks for adding the test